tests/integration/test_authd/test_api_agent_registration.py

```diff
@@ -0,0 +1,227 @@
+'''
+copyright: Copyright (C) 2015-2021, Wazuh Inc.
+
+           Created by Wazuh, Inc. <info@wazuh.com>.
+
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+type: integration
+
+brief: These tests will check if the 'wazuh-authd' daemon correctly handles the enrollment requests
+       from the API.
+
+tier: 0
+
+modules:
+    - authd
+    - API
+
+components:
+    - manager
+
+daemons:
+    - wazuh-authd
+    - wazuh-api
+
+os_platform:
+    - linux
+
+os_version:
+    - Arch Linux
+    - Amazon Linux 2
+    - Amazon Linux 1
+    - CentOS 8
+    - CentOS 7
+    - CentOS 6
+    - Ubuntu Focal
+    - Ubuntu Bionic
+    - Ubuntu Xenial
+    - Ubuntu Trusty
+    - Debian Buster
+    - Debian Stretch
+    - Debian Jessie
+    - Debian Wheezy
+    - Red Hat 8
+    - Red Hat 7
+    - Red Hat 6
+
+references:
+    - https://documentation.wazuh.com/current/user-manual/registering/restful-api-registration.html
+
+tags:
+    - authd
+    - api
+'''
+import ipaddress
+import re
+import requests
+import os
+import pytest
+import time
+from pathlib import Path
+
+from wazuh_testing.constants.paths.logs import WAZUH_API_LOG_FILE_PATH
+from wazuh_testing.constants.paths.configurations import WAZUH_CLIENT_KEYS_PATH
+from wazuh_testing.utils.file import truncate_file
+from wazuh_testing.utils.services import control_service
+from wazuh_testing.modules.api.utils import get_base_url, login
+from wazuh_testing.utils.configuration import get_test_cases_data
+
+from . import TEST_CASES_FOLDER_PATH
+
+
+# Marks
+pytestmark = [pytest.mark.linux, pytest.mark.tier(level=0), pytest.mark.server]
+
+# Configuration
+test_cases_path = Path(TEST_CASES_FOLDER_PATH, 'cases_api_agent_registration.yaml')
+test_configuration, test_metadata, test_cases_ids = get_test_cases_data(test_cases_path)
+
+client_keys_update_timeout = 1
+
+daemons_handler_configuration = {'all_daemons': True}
+
+def retrieve_client_key_entry(agent_parameters):
+    with open(WAZUH_CLIENT_KEYS_PATH) as client_keys_file:
+        client_keys_content = client_keys_file.readlines()
+        client_keys_dictionary = []
+        for entry in client_keys_content:
+            client_key_agent_list = entry.split()
+
+            client_key_agent_dict = {}
+            client_key_agent_dict.update({'id': client_key_agent_list[0], 'name': client_key_agent_list[1],
+                                          'ip': client_key_agent_list[2], 'key': client_key_agent_list[3]})
+
+            client_keys_dictionary.append(client_key_agent_dict)
+
+    desired_entries = []
+    for client_keys_entry_dict in client_keys_dictionary:
+        if agent_parameters.items() <= client_keys_entry_dict.items():
+            desired_entries.append(agent_parameters)
+    return desired_entries
+
+
+@pytest.fixture(scope='module')
```

Comment:
Fixtures should be located in `conftest.py`. What about the `truncate_client_keys_file` fixture, isn't it the same?